### PR TITLE
:sparkles: redux-form driver

### LINKED ISSUE #74: add redux-form driver
Waiting for #69 so the lib are not requiring `redux-form` everytime... for everycases.

**driver**
```js
import { setSubmitSucceeded, getFormValues, setSubmitFailed, startSubmit, stopSubmit, isSubmitting } from 'redux-form'

const asyncSubmit = (name, dispatch, getState) => async (callback, ...options) => {
  dispatch(startSubmit(name))
  await callback(...options)
  if (isSubmitting(name)(getState())) dispatch(stopSubmit)
}

export default ({ dispatch, getState }) => name => ({
  getFormValues: () => getFormValues(name, state => state.ui.form)(getState()),
  setSubmitFailed: (...fields) => dispatch(setSubmitFailed(name, ...fields)),
  setSubmitSucceeded: () => dispatch(setSubmitSucceeded(name)),
  startSubmit: () => dispatch(startSubmit(name)),
  stopSubmit: errors => dispatch(stopSubmit(name, errors)),
  asyncSubmit: (callback, ...options) => asyncSubmit(name, dispatch, getState)(callback, ...options),
})
```

**Example usage in a reaction (plugued to a listener)**
```js
export const login = reaction((action, store, { http, form }) => {
  // retrieve credentials from redux
  const signin = form('signin')
  const values = signin.getFormValues()
  // submit
  signin.asyncSubmit(http('AUTH').post, '/api/login')
})
```

### PR DESCRIPTION
**Pull request by guillaumecrespel**
fixes #74

### CODE REVIEW

File: packages/drivers/redux-form/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# @k-ramel/driver-http
+> HTTP driver for k-ramel
```
Comment (fabienjuif) [inline packages/drivers/redux-form/README.md]: rename

**Comment by fabienjuif** [comment]
I'd like this driver export all redux-form selectors (and wrap them)
And all redux-form actions (and wrap them)

**Comment by guillaumecrespel** [comment]
README.MD => https://github.com/alakarteio/k-ramel/tree/driver-redux-form/packages/drivers/redux-form

todo : 
- [ ] improve README
- [ ] build package (help wanted)

File: packages/drivers/redux-form/src/selectors.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  'isPristine',
+  'isValid',
+  'isInvalid',
+  'isSubmitting',
+  'hasSubmitSucceeded',
+  'hasSubmitFailed',
+]
+
+// selectors without parameters
+export const selectorNamesWithoutParameter = [
```
Comment (fabienjuif) [inline packages/drivers/redux-form/src/selectors.js]: `WithoutNameParameter` ?

File: packages/drivers/redux-form/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# @k-ramel/driver-redux-form
```
Comment (fabienjuif) [inline packages/drivers/redux-form/README.md]: - [ ] Reference documentation of redux-form for all available actions / selectors ?

File: packages/drivers/redux-form/README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import reduxform from '@k-ramel/driver-reduf-form'
+
+export default () => {
+  const store = createStore(
+    description,
+    {
+      listeners,
+      drivers: {
+        // by default getFormState = state => state.form
+        reduxform(/* optional : getFormState */)
```
Comment (fabienjuif) [inline packages/drivers/redux-form/README.md]: `form : reduxform(/* optional : getFormState */)`

File: packages/drivers/redux-form/README.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  )
+
+  return store
+}
+```
+
+In a reaction (See main documentation about listeners/reactions)
+```js
+/* reaction.js */
+import { reaction } from '@k-ramel/k-ramel'
```
Comment (fabienjuif) [inline packages/drivers/redux-form/README.md]: `from 'k-ramel'`

File: packages/drivers/redux-form/README.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+export default () => ({
+  form,
+})
+```
+
+In store definition (See main documentation about description)
+```js
+/* store.js */
+import { createStore } from '@k-ramel/k-ramel'
```
Comment (fabienjuif) [inline packages/drivers/redux-form/README.md]: `from 'k-ramel'`

File: packages/drivers/redux-form/README.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+import description from './description'
+import listeners from './listeners'
+import reduxform from '@k-ramel/driver-reduf-form'
```
Comment (fabienjuif) [inline packages/drivers/redux-form/README.md]: `@k-ramel/driver-redux-form`

File: packages/drivers/redux-form/README.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  return store
+}
+```
+
+In a reaction (See main documentation about listeners/reactions)
+```js
+/* reaction.js */
+import { reaction } from '@k-ramel/k-ramel'
+
+export const signin = reaction((store, action, drivers) => {
```
Comment (fabienjuif) [inline packages/drivers/redux-form/README.md]: - [ ] remove `reaction`

**Comment by fabienjuif** [comment]
- [ ] Add line of this package in root README : https://github.com/alakarteio/k-ramel

File: packages/drivers/redux-form/src/selectors.js (hunk 2 above)
Comment (guillaumecrespel) [reply packages/drivers/redux-form/src/selectors.js]: i think not, this is the names of selectors whitout parameter.

File: packages/drivers/redux-form/README.md [hunk 9]
```diff
@@ -1,7 +1,8 @@
 # @k-ramel/driver-redux-form
-> redux-form driver for k-ramel
-
+> Redux Form driver for k-ramel makes accessible all actions and selectors of Redux Form
```
Comment (fabienjuif) [inline packages/drivers/redux-form/README.md]: Be consistent with others README

```
> redux-form driver for k-ramel

All Redux Form's [actions](https://redux-form.com/7.3.0/docs/api/actioncreators.md/) and [selectors](https://redux-form.com/7.3.0/docs/api/selectors.md/) are available and wrapped.```

- Main repository: [k-ramel](https://github.com/alakarteio/k-ramel)

File: packages/drivers/redux-form/README.md [hunk 10]
```diff
@@ -41,9 +42,9 @@ export default () => {
 In a reaction (See main documentation about listeners/reactions)
 ```js
 /* reaction.js */
-import { reaction } from '@k-ramel/k-ramel'
+import { reaction } from 'k-ramel'
```
Comment (fabienjuif) [inline packages/drivers/redux-form/README.md]: remove reaction

**Comment by fabienjuif** [comment]
fixes #74

**Pull request merged**